### PR TITLE
Update some backend tests to use testCheckMountDestroyed

### PR DESCRIPTION
**Pull request by benashz**

### CODE REVIEW

File: vault/resource_terraform_cloud_secret_backend_test.go [hunk 1]
```diff
@@ -3,27 +3,26 @@ package vault
...
 
-	resourceName := "vault_terraform_cloud_secret_backend.test"
+	resourceType := "vault_terraform_cloud_secret_backend"
+	resourceName := resourceType + ".test"
 
 	resource.Test(t, resource.TestCase{
 		Providers:    testProviders,
 		PreCheck:     func() { testutil.TestAccPreCheck(t) },
-		CheckDestroy: testAccTerraformCloudSecretBackendCheckDestroy,
+		CheckDestroy: testCheckMountDestroyed(resourceType, consts.MountTypeGCP, consts.FieldBackend),
```
Comment (vinay-gopalan) [inline vault/resource_terraform_cloud_secret_backend_test.go:25]: I see that `resource_okta_auth_backend_test.go` also is using a custom Mount Check Destroy function. Should we also update that here?

Comment (benashz) [reply vault/resource_terraform_cloud_secret_backend_test.go:25]: fixing....

Comment (benashz) [reply vault/resource_terraform_cloud_secret_backend_test.go]: Fixed with: dd72c7d9e92a130346a8b4fd38407e6e7ce3897d

File: vault/resource_jwt_auth_backend_test.go [hunk 2]
```diff
@@ -304,12 +304,13 @@ resource "vault_namespace" "test" {
 }
 `, ns),
 		}
-		config += `
+		c += `
```
[the remark this answers is not in the document]
Comment (benashz) [reply vault/resource_jwt_auth_backend_test.go:308]: should be fixed with 7014c2356b595c537c10a4823a5de893e6ddba20

**Review by vinay-gopalan — APPROVED** [review]
nit: looks like we have one remaining `testAccOktaAuthBackend_Destroyed` in Okta Auth Backend tests, but looks good otherwise!

File: vault/resource_consul_secret_backend.go [hunk 3]
```diff
@@ -206,18 +210,18 @@ func consulSecretBackendRead(_ context.Context, d *schema.ResourceData, meta int
...
 	if err != nil {
 		return diag.Errorf("error reading from Vault: %s", err)
 	}
 
+	d.Set("path", path)
+	d.Set("description", mount.Description)
+	d.Set("default_lease_ttl_seconds", mount.Config.DefaultLeaseTTL)
+	d.Set("max_lease_ttl_seconds", mount.Config.MaxLeaseTTL)
+	d.Set("local", mount.Local)
+
```
Comment (robmonte) [inline vault/resource_consul_secret_backend.go:224]: 👍

**Review by robmonte — APPROVED** [review]
Consul-related changes look good to me!

**Pull request merged**